Improve CONTRIBUTING (#6605)
opnsense/core: CONTRIBUTING.md

When creating pull request, please heed the following:

* Base your code on the latest `master` branch to avoid manual merges
  If necessary, stable versions (branches `stable/<major>.<minor>`) may be updated
  by a team member with a cherry-pick after a pull-request is merged.
  This means a pull-request target is always the `master` branch.
* Code review may ensure in order to help shape your proposal
* Pull request must adhere to 2-Clause BSD licensing
* Explain the problem and your proposed solution
* If applicable cite the issue(s) number(s) in your pull-request description,
    for example `Fixes: #1234`, `Closes: #1234`, or `Ref: #1234`.
* Read [README.md](./README.md) to learn about the commands to shape your code

>   You may want to run `make style` before submitting your work.

Reviewer: ```suggestion
```

I'd leave this out to avoid cross-referencing in a loop. Pointing to readme above where the style part is explained suffices. We are not super strict and phpcb does most of the work post-merge anyway.

When creating pull request, please heed the following:

* Base your code on the latest `master` branch to avoid manual merges
* Code review by the team may occur to help you shape your proposal

> * Self review your proposal to catch mistakes and avoid delaying your proposal

Reviewer: Good. I'd say this goes as "test your proposal operationally"...

```suggestion
* Test your proposal operationally to catch mistakes and avoid merge delay
```

When creating pull request, please heed the following:

* Base your code on the latest `master` branch to avoid manual merges

> * Code review may ensure in order to help shape your proposal

Reviewer: I think this introduces a typo

```suggestion
* Code review may ensue in order to help shape your proposal
```

Author: Can we find a wording a bit less confusing for non native english readers ?

Reviewer: Sure, open for suggestions

Author: Are we trying to say that the user should self review or that the work will be reviewed ?

Reviewer: work will be reviewed if necessary/unclear

* Pull request must adhere to 2-Clause BSD licensing
* Explain the problem and your proposed solution
* If applicable cite the issue(s) number(s) in your pull-request description,
    for example `Fixes: #1234`, `Closes: #1234`, or `Ref: #1234`.
* Read [README.md](./README.md) to learn about the commands to shape your code

Stable release updates
----------------------

After merging a pull-request into the `master` branch a team member may cherry-pick
your work to update the current stable version (branches: `stable/<major>.<minor>`).

> 
> This is why pull-requests are always targeted to the `master` branch.

Reviewer: Ok now we don't need this naturally :)

```suggestion
```

> That's why pull-requests are always targetted to the `master` branch.

Reviewer: I know this feels picky but in technical documentation contractions are best avoided.

```suggestion
That is why pull-requests are always targeted to the `master` branch.
```

But I'd like to restructure this differently and just say something like "After commits are included in the master branch..." in a single paragraph instead of explaining after the fact
